substrate/frame/contracts/src/lib.rs

```diff
@@ -411,8 +411,9 @@ pub mod pallet {
 			use migration::MigrateResult::*;
 
 			loop {
-				let (result, weight) = Migration::<T>::migrate(remaining_weight);
-				remaining_weight.saturating_reduce(weight);
+				const reduced_weight_factor: u64 = 4;
+				let (result, weight) = Migration::<T>::migrate(remaining_weight.saturating_div(reduced_weight_factor));
+				remaining_weight.saturating_reduce(weight.saturating_mul(reduced_weight_factor));
 
 				match result {
 					// There is not enough weight to perform a migration, or make any progress, we
```

Comment:
I think that this is actually not correct to `return` from `on_idle` in the first case of the `match` expression - we should try making some other work (like `process_deletion_queue`) - does this sound reasonable?

Reply:
You are probably right, we could `break` instead, but it doesn't matter much anyway, as in practice the migration take just a few blocks of time